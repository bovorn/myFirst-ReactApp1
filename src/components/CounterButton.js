import React, { Component } from 'react';
import randomColor from '../utils/randomColor';

class CounterButton extends Component {
    constructor(props) {
        super(props);
        this.state = { 
            count: 0,
            color: this.props.color
        };
    }

    shouldComponentUpdate = (nextProps, nextState) => {
        if(this.state.count !== nextState.count){
            this.setState({color: randomColor()})
            return true;
        }
        return false;
    }
      
    render() { 

        const styles = {
            counterBtn: {
                color: this.state.color,
                backgroundColor: 'transparent',
                border: `1px solid ${this.state.color}`,
                margin: 8,
                padding: 10,
                borderRadius: 5,
                cursor: 'pointer',
            }
        }
        return (
            <button
                style={styles.counterBtn}
                id="counter"
                onClick={() => this.setState(state => ({count: state.count + 1}))}>
                Count: {this.state.count}
            </button>
        )
    }
}

export default CounterButton;
