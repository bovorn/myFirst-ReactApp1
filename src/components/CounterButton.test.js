import renderer from 'react-test-renderer';
import React from 'react';
import CounterButton from './CounterButton';

it('expect to render CounterButton component', () => {
    const wrapper = renderer.create(<CounterButton />);
    expect(wrapper.toJSON()).toMatchSnapshot();
})

it('correctly increments the counter', () => {
    const wrapper = renderer.create(<CounterButton color="red" />);
    
    wrapper.root.findByProps({id: "counter"}).props.onClick();
    expect(wrapper.getInstance().state).toEqual({ count: 1 });
    expect(wrapper.getInstance().props.color).toEqual("red");
}) 