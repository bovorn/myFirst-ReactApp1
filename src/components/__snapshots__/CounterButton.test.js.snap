// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
<button
  id="counter"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "1px solid undefined",
      "borderRadius": 5,
      "color": undefined,
      "cursor": "pointer",
      "margin": 8,
      "padding": 10,
    }
  }
>
  Count: 
  0
</button>
`;
